### ISSUE TITLE
Use pypy or brython implementation for builtins and standard lib

### ISSUE THREAD

**Issue by wolfv** [pybee/batavia#13]
Hi, 

I was wondering if it makes sense to use an implementation for the python builtins that's actually implemented in python. 

For example, pypy has the following https://bitbucket.org/pypy/pypy/src/35c27789e35368ac3ebefedd83af0829d2cf89d4/pypy/module/__builtin__/

Brython similarly has pure python implementations for large parts of the standards library (not sure about builtins, though).

Maybe it would be good to create a distinguished repo for this kind of reimplementation of python in pure python.

Cheers,

Wolf

**Comment by freakboy3742** [comment]
Yes - I think it would. Reusing an existing builtins + standard library would be desirable; collaborating on a common "pure python" repo would be even better. In addition to Batavia, [VOC](https://github.com/pybee/voc) has a similar need for a pure Python stdlib implementation.

Having an clear specification for the interface that a native implementation must provide, and a way to override specific modules to provide a native interface would also be desirable.

I'm happy to host the project as part of BeeWare; I have no idea whether the PyPy or Brython teams would be interested in using it. In an ideal world, it would be something that is shared with CPython as well - but that's probably a pipe dream.

**Comment by freakboy3742** [comment]
Some possible names for the project:

* Troupe (as in, a collection of (Monty) Pythons)
* Pythonidae (the taxonomic name for Pythons)
* Clutch (as in, a collection of Python eggs)
* Nest (as in, a nest of vipers)

**Comment by wolfv** [comment]
My working draft name was pyinpy but it's probably too close to pypy, hehe.

I think pythonidae is quite cool, but I am happy with any of them.

Cheers,

Wolf

**Comment by funkybob** [comment]
Surely "Ouroboros"? :)

**Comment by freakboy3742** [comment]
@funkybob Oh... oh my.... We have a winner. Ouroboros it is.

[Registered with PyPI](https://pypi.python.org/pypi/ouroboros) and [repository created](https://github.com/pybee/ouroboros).
